oauth2-native/src/main/java/org/ballerinalang/stdlib/oauth2/OAuth2Client.java

```diff
@@ -132,26 +160,132 @@ public void checkServerTrusted(X509Certificate[] certs, String authType) {
+        KeyStore ts = buildTrustStore(cert);
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init(ts);
+        return buildSslContext(null, tmf.getTrustManagers());
+    }
+
+    private static SSLContext initSslContext(BMap<BString, BString> trustStore, BMap<BString, BString> keyStore)
+            throws Exception {
+        BString trustStorePath = trustStore.getStringValue(OAuth2Constants.PATH);
+        BString trustStorePassword = trustStore.getStringValue(OAuth2Constants.PASSWORD);
+        KeyStore ts = getKeyStore(trustStorePath, trustStorePassword);
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        tmf.init(ts);
+        BString keyStorePath = keyStore.getStringValue(OAuth2Constants.PATH);
+        BString keyStorePassword = keyStore.getStringValue(OAuth2Constants.PASSWORD);
+        KeyStore ks = getKeyStore(keyStorePath, keyStorePassword);
+        KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        kmf.init(ks, keyStorePassword.getValue().toCharArray());
+        return buildSslContext(kmf.getKeyManagers(), tmf.getTrustManagers());
+    }
+
+    private static SSLContext initSslContext(BMap<BString, BString> trustStore, BString certFile, BString keyFile,
+                                             BString keyPassword) throws Exception {
+        BString trustStorePath = trustStore.getStringValue(OAuth2Constants.PATH);
+        BString trustStorePassword = trustStore.getStringValue(OAuth2Constants.PASSWORD);
+        KeyStore ts = getKeyStore(trustStorePath, trustStorePassword);
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        tmf.init(ts);
+        KeyStore ks = buildKeyStore(certFile, keyFile, keyPassword);
+        KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        kmf.init(ks, "".toCharArray());
+        return buildSslContext(kmf.getKeyManagers(), tmf.getTrustManagers());
+    }
+
+    private static SSLContext initSslContext(BString cert, BString certFile, BString keyFile,
+                                             BString keyPassword) throws Exception {
+        KeyStore ts = buildTrustStore(cert);
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init(ts);
+        KeyStore ks = buildKeyStore(certFile, keyFile, keyPassword);
+        KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        kmf.init(ks, "".toCharArray());
+        return buildSslContext(kmf.getKeyManagers(), tmf.getTrustManagers());
+    }
+
+    private static SSLContext initSslContext(BString cert, BMap<BString, BString> keyStore) throws Exception {
```

Comment:
Isn't it better to have separate functions like 
1. return TrustManagerFactory giving a cert
2. return TrustManagerFactory giving a truststore
3. return KeyManagerFactory giving keys and certs
4. return KeyManagerFactory giving keystore

And then call `buildSslContext` passing them?

Reply:
That's better. Fixed with https://github.com/ballerina-platform/module-ballerina-oauth2/pull/98/commits/5b431622d20737c03282896a2589834aba4159ba